cmake/flexdef.h

```diff
@@ -0,0 +1,1249 @@
+
+/* flexdef - definitions file for flex */
+
+/*  Copyright (c) 1990 The Regents of the University of California. */
+/*  All rights reserved. */
+
+/*  This code is derived from software contributed to Berkeley by */
+/*  Vern Paxson. */
+
+/*  The United States Government has rights in this work pursuant */
+/*  to contract no. DE-AC03-76SF00098 between the United States */
+/*  Department of Energy and the University of California. */
+
+/*  This file is part of flex. */
+
+/*  Redistribution and use in source and binary forms, with or without */
+/*  modification, are permitted provided that the following conditions */
+/*  are met: */
+
+/*  1. Redistributions of source code must retain the above copyright */
+/*     notice, this list of conditions and the following disclaimer. */
+/*  2. Redistributions in binary form must reproduce the above copyright */
+/*     notice, this list of conditions and the following disclaimer in the */
+/*     documentation and/or other materials provided with the distribution. */
+
+/*  Neither the name of the University nor the names of its contributors */
+/*  may be used to endorse or promote products derived from this software */
+/*  without specific prior written permission. */
+
+/*  THIS SOFTWARE IS PROVIDED ``AS IS'' AND WITHOUT ANY EXPRESS OR */
+/*  IMPLIED WARRANTIES, INCLUDING, WITHOUT LIMITATION, THE IMPLIED */
+/*  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR */
+/*  PURPOSE. */
+
+#ifndef FLEXDEF_H
+#define FLEXDEF_H 1
+
+#ifdef _MSC_VER 
+
+#include "sys/stat.h"
+#include <stdio.h>
+#include <stdint.h>
+
+/* Avoid Windows.h - obviously only works for host builds... */
+uint32_t htonl(uint32_t x);
+uint16_t htons(uint16_t x);
+
+
+#include <io.h>
+#define close _close
+#define dup2 _dup2
+#define dup _dup
+#define fileno _fileno
+#define fdopen _fdopen
+#define unlink _unlink
+#define strdup _strdup
+#define isatty _isatty
+
+/* Stubs rfom sys/stat.h */
+#define S_IXUSR 0x0100 // Execute/search permission, owner.
+#define S_ISREG(m) (1)
+
+/* Stubs from missing sys/wait.h */
+#ifdef _POSIX_SOURCE
+#define	_W_INT(i)	(i)
+#else
+#define	_W_INT(w)	(*(int *)&(w))	/* convert union wait to int */
+#endif
+
+#define	_WSTATUS(x)	(_W_INT(x) & 0177)
+#define WIFEXITED(x)	(_WSTATUS(x) == 0)
+#define WEXITSTATUS(x)	(_W_INT(x) >> 8)
+
+// return program name...
+#define _MAX_NODE _MAX_DIR
+
+#ifdef INT8_MIN
+#undef INT8_MIN
+#endif
+#ifdef INT16_MIN
+#undef INT16_MIN
+#endif
+#ifdef INT32_MIN
+#undef INT32_MIN
+#endif
+#ifdef INT8_MAX
+#undef INT8_MAX
+#endif
+#ifdef INT16_MAX
+#undef INT16_MAX
+#endif
+#ifdef INT32_MAX
+#undef INT32_MAX
+#endif
+#ifdef UINT8_MAX
+#undef UINT8_MAX
+#endif
+#ifdef UINT16_MAX
+#undef UINT16_MAX
+#endif
+
+
+#endif //_WIN32
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
```

Comment:
Principally `config.h` should be included before other standard headers, as `config.h` would contain feature macros that would alter which function interface that standard headers would expose. Is there any reason you include the other way around?

Reply:
Yes see above comment.  I appreciate the time you are putting into reviewing this :)